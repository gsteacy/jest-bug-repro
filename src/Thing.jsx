import * as React from "react";

export default function Thing({ className, children }) {
    return (
        <div className={className}>
            {children}
        </div>
    );
}