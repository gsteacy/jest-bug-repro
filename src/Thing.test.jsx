import * as React from "react";
import * as renderer from 'react-test-renderer';

import Thing from "./Thing";

describe("tests", () => {
    test("defaults", () => {
        expect(render(<Thing />)).toMatchSnapshot();
    });

    test("class", () => {
        expect(render(<Thing className="a class" />)).toMatchSnapshot();
    });

    // test("children", () => {
    //     expect(render(<Thing><div /></Thing>)).toMatchSnapshot();
    // });
});

function render(component) {
    return renderer.create(component).toJSON();
}