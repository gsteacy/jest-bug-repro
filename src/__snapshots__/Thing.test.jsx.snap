// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests class 1`] = `
<div
  className="a class"
/>
`;

exports[`tests defaults 1`] = `
<div
  className={undefined}
/>
`;
